src/audiovideocontroller/DefaultAudioVideoController.ts

```diff
@@ -1103,18 +1110,25 @@ export default class DefaultAudioVideoController
       this.meetingSessionContext.videoUplinkBandwidthPolicy &&
       !this.meetingSessionContext.enableSimulcast
     ) {
-      const oldMaxBandwidth = this.meetingSessionContext.videoUplinkBandwidthPolicy.maxBandwidthKbps();
-
-      this.meetingSessionContext.videoUplinkBandwidthPolicy.setHasBandwidthPriority(
-        hasBandwidthPriority
-      );
-      const newMaxBandwidth = this.meetingSessionContext.videoUplinkBandwidthPolicy.maxBandwidthKbps();
+      if (this.meetingSessionContext.videoUplinkBandwidthPolicy.chooseEncodingParameters().size > 0) {
+        this.meetingSessionContext.videoUplinkBandwidthPolicy.setHasBandwidthPriority(
+          hasBandwidthPriority
+        );
+        await this.enforceBandwidthLimitationForSender(this.meetingSessionContext.videoUplinkBandwidthPolicy.maxBandwidthKbps());
+      } else {
+        const oldMaxBandwidth = this.meetingSessionContext.videoUplinkBandwidthPolicy.maxBandwidthKbps();
 
-      if (oldMaxBandwidth !== newMaxBandwidth) {
-        this.logger.info(
-          `video send bandwidth priority ${hasBandwidthPriority} max has changed from ${oldMaxBandwidth} kbps to ${newMaxBandwidth} kbps`
+        this.meetingSessionContext.videoUplinkBandwidthPolicy.setHasBandwidthPriority(
+          hasBandwidthPriority
         );
-        await this.enforceBandwidthLimitationForSender(newMaxBandwidth);
+        const newMaxBandwidth = this.meetingSessionContext.videoUplinkBandwidthPolicy.maxBandwidthKbps();
```

Comment:
This is a bit tricky. I'd _like_ to say that if the policy is returning encoding parameters, we don't also need to special-case max bandwidth, and avoid having this code.